src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -70,6 +70,8 @@ internal enum XmlDocCommentStyle
 
     internal sealed partial class Lexer : AbstractLexer
     {
+        private static readonly ObjectPool<LexerCache> s_lexerCachePool = new ObjectPool<LexerCache>(() => new LexerCache());
```

Comment:
>s_lexerCachePool [](http://example.com/codeflow?start=55&length=16)

Since we are making ```LexerCache``` pollable, would it make sense to encapsulate its pool in ```LexerCache```?